scripts/generate_version.py

```diff
@@ -0,0 +1,39 @@
+import json
+import subprocess
+import sys
+from time import strftime
+
+
+def getVersionInfo():
+    now = strftime("%Y-%m-%d %H:%M:%S")
+    # If we would like the short form, add '--short' option
+    # sha = subprocess.check_output(['git', 'rev-parse', '--short', 'HEAD'])
+    sha = subprocess.check_output(["git", "rev-parse", "HEAD"])
+    # For Python3, convert bytes to str
+    if isinstance(sha, bytes):
+        sha = sha.decode()
+    sha = sha.strip()
+    version = {"date": now, "sha": sha}
+    json_string = json.dumps(version, indent=4)
+    return json_string
+
+
+def showUsage():
+    print("Usage: python[3] %s output_file" % sys.argv[0])
+
+
+def writeFile(path, string):
+    f = open(path, "w")
+    f.write(string)
+    f.close()
+
+
+if __name__ == "__main__":
+    if len(sys.argv) == 2:
+        file_path = sys.argv[1]
+    else:
+        showUsage()
+        exit()
```

Comment:
There are some other CI error about this.
Can you run `make` to see if you can pass all tests locally?

Reply:
Sounds the new error come from pylint.  Anyway, will update according to your suggestion later.